src/main/proto/cosmas.proto

```diff
@@ -37,9 +37,34 @@ message CreateVersionRequest {
 message CreateVersionResponse {
 }
 
+message CreateHistoryOfVersionRequest {

```

Comment:
When you type a few words in the editor and send a small patch to Cosmas, you do not create a new version, because version is the whole file text, not a single patch. You do not create a new history either, because history by definition is an (order) sequence of some other actions. 
What you do when you send a small patch you, well, create a new patch. 

Let's name the things appropriately. When I read "CreateHistoryOfVersion" I think that this method creates a history of versions. When I learn that actually it adds a patch, I am surprised and my WTF/second rate grows. When you write code your goal (besides writing  code that works of course) is to minimize that WTF/second rate for those who will read your code.